docs/src/GettingStarted/Terminology.md

```diff
@@ -0,0 +1,78 @@
+# Terminology
+
+The `ClimateMachine` documentation uses terminology from several disciplines. Below are some definitions of some of these terms. Please let us know via a comment if there are others that you find confusing, and we will add them.
+
+## Software terms
+
+* __Driver file__: a julia script which (a) specifies which
+[`BalanceLaw`](../APIs/BalanceLaws/BalanceLaws.md#ClimateMachine.BalanceLaws.BalanceLaw)
+is being used, (b) specifies changes to the default version
+of that model, (c) specifies initial and boundary conditions,
+(d) specifies numerical details related to timestep, grid,
+and frequency of output, and (e) runs the integration.
+Examples can be found in tutorials and experiments.
+* __Source file__: a julia file containing code involved
+in implementing the choices made in a driver file. This
+is where the numerical methods related to solving the
+equations (e.g. the ODE time steppers, or the creation of
+the grid) are defined. Users typically only interact with
+functions defined in source code within a driver file by
+using arguments to those functions, unless they wish to
+develop the source code itself.
+* __Kernel__: Functions which are launched on the compute device (i.e., the CPU or
+GPU) and run with an array of workitems or threads.
+`ClimateMachine.jl` uses
+[`KernelAbstractions.jl`](https://github.com/JuliaGPU/KernelAbstractions.jl)
+as its kernel language.
+* __Callback__: Functions executed by the ODE integrator after each time step; see
+[`solve!`](#ClimateMachine.ODESolvers.solve!).
+This allows the ability to inject custom behavior into the ODE
+integrators, such as diagnostic output, visualization, and apply
+filtering to the numerical solution.
+Several convenience functions exists specifying
+[callback](../APIs/Numerics/ODESolvers/ODESolvers.md/#ClimateMachine.ODESolvers.GenericCallback)
```

Comment:
Not sure why this link is not working. It copies what does work, for the BalanceLaw API.